sdks/go/pkg/beam/core/funcx/output.go

```diff
@@ -84,6 +84,9 @@ func unfoldEmit(t reflect.Type) ([]reflect.Type, bool, error) {
 		if ok, err := isInParam(t.In(i)); !ok {
 			return nil, false, errors.Wrap(err, errIllegalParametersInEmit)
 		}
+		if t.In(i).Kind() == reflect.Interface || (t.In(i).Kind() == reflect.Pointer && t.In(i).Elem().Kind() == reflect.Interface) {
+			panic("Type interface{} isn't a supported PCollection type")
```

Comment:
If there's an error return, one should return an error instead of panicking. Panicking is generally a last resort thing, and rarely when one can return an error instead.